internal/rootcoord/meta_table.go

```diff
@@ -550,12 +550,12 @@ func (mt *MetaTable) RemoveCollection(ctx context.Context, collectionID UniqueID
 }
 
 func filterUnavailable(coll *model.Collection) *model.Collection {
-	clone := coll.Clone()
+	clone := coll.ShadowClone()
```

Comment:
Talked offline, need some read only interface to protect the shallow copy from unintended modification

Reply:
Agree, and I will implement it in the next PR.